### DISCUSSION TITLE
Add `highlight` to `kedro-viz` section of `dataset.metadata` to highlight it in the UI

### DISCUSSION THREAD

**Discussion by yury-fedotov** [kedro-org/kedro-viz#1852] [Ideas]
## Description
Add **`highlight`** to `kedro-viz` section of `dataset.metadata` to highlight it in the UI as a "key dataset" - e.g. with **yellow background fill and black text**.

## Context
I'm regularly walking the team / audience through the `kedro viz` UI and almost always, on this huge DAG representation there are 2-3 datasets which are really the key artifacts I explain the flow in respect to.

Usually those are:
* Master table
* Trained model

Using modular pipelines, and hence leveraging the collapsing supernodes, is incredibly helpful is simplifying the DAG view and visually tracking where those key artifacts are. However, I though that since the metadata injection logic already exists in the `kedro viz` (e.g. adding `layer` to datasets), I thought that if the group decides that this feature is useful, it might be relatively easy to implement.

## Possible Implementation
So the most elegant way I see is to do this exactly like e.g. `layer` can be added to a dataset in its `catalog` definition.

For example, let's say the `trained_model` is the one I want to highlight. See where I'm adding `highlight: true`:

```yaml
# catalog.yml

trained_model:
  type: ...
  filepath ...
  metadata:
    kedro-viz:
      layer: models
      highlight: true
```

## Possible Alternatives
Probably the above is the most natural place for this functionality to live, so not sure.

## Checklist

- [x] Include labels so that we can categorise your feature request

**Comment by astrojuanlu** [comment]
Thanks for opening this issue @yuryfedotov-mck !

@rashidakanchwala @NeroOkwa I think this is somewhat related to #1148, adds support for my idea of having a `metadata.kedro-viz.style` key (or even... support custom CSS????)
